### DISCUSSION TITLE
xUnit & making sure that `ConfigureAwait(true)` is used on the top levels

### DISCUSSION THREAD

**Discussion by MartyIX** [xunit/xunit#2210] [Question] [answered]
Hi xUnit team,

I wonder whether there is an xUnit analyzator that would make sure that all my top-level `async` method calls use only `ConfigureAwait(true)`. Is there?

Thank you!

**Comment by bradwilson** [comment, marked as answer]
There isn't. Given that this is not strictly a testing issue, but rather a production code issue, we feel that this analyzer probably belongs elsewhere (and I'm sure there are several available if you search for them). Our analyzers only run on test projects, and are targeted only for issues related to our test framework.

Thanks!
